### PR TITLE
feat: add useKeyDown hook

### PR DESCRIPTION
**Pull request by Kritik-J**
Resolves: #568

### CODE REVIEW

File: frontend/webapp/hooks/useKeyDown.tsx [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+export function useKeyDown(key, callback) {
+  const handleKeyDown = (event) => {
+    if (key === event.key) {
+      callback(event);
+    }
+  };
+
+  useEffect(() => {
+    window.addEventListener("keydown", handleKeyDown);
```
Comment (alonkeyval) [inline frontend/webapp/hooks/useKeyDown.tsx:11]: keypress event is a deprecated feature and is no longer recommended. please replace the event with the relevant one

File: frontend/webapp/hooks/useKeyDown.tsx [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+    return () => {
+      window.removeEventListener("keydown", handleKeyDown);
+    };
+  }, [key, callback]);
+
+  return handleKeyDown;
+}
```
Comment (alonkeyval) [inline frontend/webapp/hooks/useKeyDown.tsx:19]: please change the file to .ts instead of .tsx

**Comment by alonkeyval** [comment]
@Kritik-J  please implement the new hook in the relevant files in addition to the hook implementation

File: frontend/webapp/hooks/useKeyDown.tsx (hunk 1 above)
Comment (Kritik-J) [reply frontend/webapp/hooks/useKeyDown.tsx:11]: keypress is deprecated, so i used keydown instead

image

**Review by alonkeyval — APPROVED** [review]
LGTM

**Comment by blumamir** [comment]
@Kritik-J - the CI seems to fail, can you please take a look?
Thanks

```
#21 39.30 1:44  Error: Strings must use singlequote.  quotes
#21 39.30 7:8  Error: Strings must use singlequote.  quotes
#21 39.30 14:8  Error: Strings must use singlequote.  quotes
#21 39.30 15:30  Error: Strings must use singlequote.  quotes
#21 39.30 16:23  Error: Strings must use singlequote.  quotes
#21 39.30 17:33  Error: Strings must use singlequote.  quotes
#21 39.30 18:23  Error: Strings must use singlequote.  quotes
#21 39.30 19:19  Error: Strings must use singlequote.  quotes
#21 39.30 20:28  Error: Strings must use singlequote.  quotes
#21 39.30 40:9  Error: Strings must use singlequote.  quotes
#21 39.30 41:9  Error: Strings must use singlequote.  quotes
#21 39.30 42:9  Error: Strings must use singlequote.  quotes
#21 39.30 54:29  Error: Strings must use singlequote.  quotes
```

**Comment by Kritik-J** [comment]
@blumamir please review

**Pull request merged**